tools/test-case-generators/genall.go

```diff
@@ -0,0 +1,515 @@
+	data := struct {
+		ComposeRequest composeRequest                 `json:"compose-request"`
+		Manifest       distro.Manifest                `json:"manifest"`
+		RPMMD          map[string][]rpmmd.PackageSpec `json:"rpmmd"`
+		NoImageInfo    bool                           `json:"no-image-info"`
+	}{
+		cr, manifest, pkgs, true,
+	}
+	b, err := json.MarshalIndent(data, "", "  ")
+	if err != nil {
+		return fmt.Errorf("failed to marshal data for %q: %s\n", filename, err.Error())
+	}
+	b = append(b, '\n') // add new line at end of file
+	fpath := filepath.Join(path, filename)
+	fp, err := os.Create(fpath)
+	if err != nil {
+		return fmt.Errorf("failed to create output file %q: %s\n", fpath, err.Error())
+	}
+	defer fp.Close()
+	if _, err := fp.Write(b); err != nil {
+		return fmt.Errorf("failed to write output file %q: %s\n", fpath, err.Error())
+	}
+	return nil
+}
+
+func filterRepos(repos []repository, typeName string) []repository {
+	filtered := make([]repository, 0)
+	for _, repo := range repos {
+		if len(repo.ImageTypeTags) == 0 {
+			filtered = append(filtered, repo)
+		} else {
+			for _, tt := range repo.ImageTypeTags {
+				if tt == typeName {
+					filtered = append(filtered, repo)
+					break
+				}
+			}
+		}
+	}
+	return filtered
+}
+
+// collects requests from a formatRequestMap based on image type
+func requestsByImageType(requestMap formatRequestMap) map[string]map[string]manifestRequest {
+	imgTypeRequestMap := make(map[string]map[string]manifestRequest)
+
+	for name, req := range requestMap {
+		it := req.ComposeRequest.ImageType
+		reqs := imgTypeRequestMap[it]
+		if reqs == nil {
+			reqs = make(map[string]manifestRequest)
+		}
+		reqs[name] = req
+		imgTypeRequestMap[it] = reqs
+	}
+	return imgTypeRequestMap
+}
+
+type workerQueue struct {
+	// manifest job channel
+	jobQueue chan manifestJob
+
+	// channel for sending messages from jobs to the printer
+	msgQueue chan string
+
+	// channel for sending errors from jobs to the collector
+	errQueue chan error
+
+	// global error list
+	errors []error
+
+	// total job count defined on workerQueue creation
+	// sets the length of the job queue so that pushing to the queue doesn't block
+	njobs uint32
+
+	// total workers defined on workerQueue creation
+	nworkers uint32
+
+	// active worker count
+	activeWorkers int32
+
+	// wait group for all workers
+	workerWG sync.WaitGroup
+
+	// wait group for internal routines (printer and error collector)
+	utilWG sync.WaitGroup
+}
+
+func newWorkerQueue(nworkers uint32, njobs uint32) *workerQueue {
+	wq := workerQueue{
+		jobQueue:      make(chan manifestJob, njobs),
+		msgQueue:      make(chan string, nworkers),
+		errQueue:      make(chan error, nworkers),
+		errors:        make([]error, 0, nworkers),
+		nworkers:      nworkers,
+		activeWorkers: 0,
+		njobs:         njobs,
+	}
+
+	return &wq
+}
+
+func (wq *workerQueue) start() {
+	wq.startMessagePrinter()
```

Comment:
I was thinking about this, and I realized that we might just want to do the message printing and error collection on the main thread to save a bit of synchronization and 2 goroutines. Not sure if worth it tbh, just an idea.

Reply:
Could rework it in a followup if you don't mind keeping it for now.  There's certainly room for improvement in a few places for the generator.  Especially for code quality.